addon/components/mobile/flexberry-objectlistview.js

```diff
@@ -138,4 +140,159 @@ export default FlexberryObjectlistview.extend({
     @default true
   */
   columnsWidthAutoresize: true,
+  /**
+    Array of objects corresponding to list of pages.
+
+    Each page is presented as object with following properties:
+    - **number** - Number of page.
+    - **isCurrent** - Page is current.
+    - **isEllipsis** - If `true` this page not showing in list.
+
+    @property _pages
+    @type Array
+    @readOnly
+  */
+
+  mobilePages: Ember.computed('pages', function() {
+    let mobilePages = [];
+    let pages = this.get('pages');
+    let currentPageNumber = pages.find(page => page.isCurrent).number;
+
+    // Rebuilding pages for mobile version.
+    switch (currentPageNumber) {
+      case 1:
+      case 2:
+        mobilePages.push(pages[0], pages[1], pages[2]);
+        break;
+      case 3:
+        mobilePages.push(pages[1], pages[2], pages[3]);
+        break;
+      case pages[pages.length - 2].number:
+      case pages[pages.length - 1].number:
+        mobilePages.push(pages[pages.length - 3], pages[pages.length - 2], pages[pages.length - 1]);
+        break;
+      default:
+        mobilePages.push(pages[2], pages[3], pages[4]);
+        break;
+    }
+
+    return mobilePages;
+  }),
+
+  /**
+    Array of pages without current.
+
+    @property _allPages
+    @type Array
+    @readOnly
+  */
+  _allPages: Ember.computed('pages', function() {
```

Comment:
Думаю что зависимость должна быть от `mobilePages`, раз на его основе вычисляется.
```suggestion
  _allPages: Ember.computed('mobilePages', function() {
```